### ISSUE TITLE
Odin: Quick glimpse at unlocked Desktop possible after resume

### ISSUE THREAD

**Issue by marbetschar** [elementary/greeter#479]
## What Happened

When my laptop suspends the screen goes dark. So far everythings fine. But when I wake it up again, I can get a quick glimpse at the unlocked Desktop before the Greeter is shown.

## Expected Behavior

The only thing visible after resume should be the Greeter, so a user must login before they can see anything of the Desktop.

## Steps to Reproduce

1. Boot and login as usual
2. Send the laptop to sleep
3. Resume the laptop and you'll get a quick glimpse at the Desktop at the state it had before hibernation
4. The Greeter is shown and hides the Desktop

## Platform Information

* Using elementary Odin. But if I recall correctly this was also a thing on Hera.

**Comment by hanaral** [comment]
@cassidyjames sorry for tagging you like this, but I wanted to know your thoughts on the current late-locking function. 
It seems to introduce a few security issues like the ones listed above....

**Comment by cassidyjames** [comment]
@hanaral I recall seeing this on earlier versions of elementary OS without the new screen shield, so I'm not sure this is new. @davidmhewitt might have thoughts on how to reduce the flash of unlocked content, but I'm not sure.

**Comment by davidmhewitt** [comment]
Yeah it's actually improved for me with the new screen shield. With light locker, I got 2-5 seconds of my unlocked session before the unlock screen, now it's just a split second. But it seems to be hardware dependant.

It might be possible to improve on it. Next time I'm digging in that area, I'll take a look.

**Comment by hanaral** [comment]
Alright, but I hope you guys are also aware that the late-locking black screen can allow for system sounds to play, and for the viewing of anything that draws over everything else (like Steam private messages, and I'm sure there are more). I _really_ think that locking should go directly to the greeter...

**Comment by davidmhewitt** [comment]
>  and for the viewing of anything that draws over everything else (like Steam private messages, and I'm sure there are more).

Well that sounds like a bug that can be fixed, I thought I'd seen a separate bug about that somewhere that I can't find now.
